src/main/resources/pdl/schema.graphql

```diff
@@ -0,0 +1,91 @@
+    originaltNavn: OriginaltNavn
+    gyldigFraOgMed: Date
+    folkeregistermetadata: Folkeregistermetadata
+    metadata: Metadata!
+}
+
+type OriginaltNavn {
+    fornavn: String
+    mellomnavn: String
+    etternavn: String
+}
+
+type Folkeregistermetadata {
+    ajourholdstidspunkt: DateTime
+    gyldighetstidspunkt: DateTime
+    opphoerstidspunkt: DateTime
+    kilde: String
+    aarsak: String
+    sekvens: Int
+}
+
+type Metadata {
+    # I PDL så får alle forekomster av en opplysning en ID som representerer dens unike forekomst.
+    # F.eks, så vil en Opprett ha ID X, korriger ID Y (der hvor den spesifiserer at den korrigerer X).
+    # Dersom en opplysning ikke er lagret i PDL, så vil denne verdien ikke være utfylt.
+    opplysningsId: String
+
+    # Master refererer til hvem som eier opplysningen, f.eks så har PDL en kopi av Folkeregisteret, da vil master være FREG og eventuelle endringer på dette må gå via Folkeregisteret (API mot dem eller andre rutiner).
+    master: String!
+
+    # En liste over alle endringer som har blitt utført over tid.
+    # Vær obs på at denne kan endre seg og man burde takle at det finnes flere korrigeringer i listen, så dersom man ønsker å kun vise den siste, så må man selv filtrere ut dette.
+    # Det kan også ved svært få tilfeller skje at opprett blir fjernet. F.eks ved splitt tilfeller av identer. Dette skal skje i svært få tilfeller. Dersom man ønsker å presentere opprettet tidspunktet, så blir det tidspunktet på den første endringen.
+    endringer: [Endring!]!
+
+    # Feltet betegner hvorvidt dette er en funksjonelt historisk opplysning, for eksempel en tidligere fraflyttet adresse eller et foreldreansvar som er utløpt fordi barnet har fylt 18 år.
+    # I de fleste tilfeller kan dette utledes ved å se på de andre feltene i opplysningen. Dette er imidlertid ikke alltid tilfellet, blant annet for foreldreansvar.
+    # Feltet bør brukes av konsumenter som henter informasjon fra GraphQL med historikk, men som også trenger å utlede gjeldende informasjon.
+    historisk: Boolean!
+}
+
+type Endring {
+    # Hvilke type endring som har blitt utført.
+    type: Endringstype!
+    # Tidspunktet for registrering.
+    registrert: DateTime!
+    # Hvem endringen har blitt utført av, ofte saksbehandler (f.eks Z990200), men kan også være system (f.eks srvXXXX). Denne blir satt til "Folkeregisteret" for det vi får fra dem.
+    registrertAv: String!
+    # Hvilke system endringen har kommet fra (f.eks srvXXX). Denne blir satt til "FREG" for det vi får fra Folkeregisteret.
+    systemkilde: String!
+    # Opphavet til informasjonen. I NAV blir dette satt i forbindelse med registrering (f.eks: Sykehuskassan).
+    # Fra Folkeregisteret får vi opphaven til dems opplysning, altså NAV, UDI, Politiet, Skatteetaten o.l.. Fra Folkeregisteret kan det også være tekniske navn som: DSF_MIGRERING, m.m..
+    kilde: String!
+}
+
+enum Endringstype {
+    OPPRETT
+    KORRIGER
+    OPPHOER
+}
```

Comment:
Hva brukes denne schema-filen til?

Reply:
I utgangspunktet er schema-filen ikke nødvendig for at det spørringen skal fungere, men den fungerer som et hjelpemiddel. Den definerer hvilke elementer som er lovlige i en graphQL-spørring til PDL. (jeg har selvfølgelig ikke definert hele PDL-schemaet, da dette ble litt i overkant). Tar man vekk schema-filen vil det man har definert i person.graphql bli markert med rødt av Intellij fordi den ikke finner definisjoner av typene man bruker. Ideen med denne schema-filen er tatt fra slik de har gjort det i pesys.